src/game_inventory.cpp

```diff
@@ -384,6 +387,32 @@ class comestible_inventory_preset : public inventory_selector_preset
                     return to_string_clipped( spoils );
                 }
                 return std::string();
+            }, _( "SHELF LIFE" ) );
+
+            append_cell( [this]( const item_location & loc ) {
+                if( g->u.get_skill_level( skill_cooking ) >= 3 ||
+                    g->u.get_skill_level( skill_survival ) >= 4 ) {
+                    const islot_comestible item = get_edible_comestible( loc );
+                    if( item.spoils > 0 ) {
+                        std::string freshness = get_freshness( loc );
```

Comment:
Is there a reason you store this in a local variable instead of returning it directly?

Reply:
As I recall, it's for de-constifying, but for a very annoying reason -- we also return a plain `new std::string()` below, which is definitely non-const, and a function obviously can't have two return types...

There must be a better solution, but I'm not a C++ person and the google isn't helping. How do I get the `new std::string()` to be const?

Reply:
It should just work to `return get_freshness(loc)`.  I tried checking out your branch and that change compiles for me.

Reply:
> As I recall, it's for de-constifying, but for a very annoying reason -- we also return a plain new std::string() below, which is definitely non-const, and a function obviously can't have two return types...

> There must be a better solution, but I'm not a C++ person and the google isn't helping. How do I get the new std::string() to be const?

`return new std::string()` returns a pointer (to `std::string`). `return std::string()` returns a string value, neither is const.

An existing lambda is this:
```C++
append_cell( [ this ]( const item_location & loc ) {
    const time_duration spoils = get_edible_comestible( loc ).spoils;
    if( spoils > 0 ) {
        return to_string_clipped( spoils );
    }
    return std::string();
}, _( "SPOILS IN" ) );
```
It returns a new string value, or a string value from function call, basically e the same as in your code.

Reply:
@BevapDin Yes, but `to_string_clipped` does not return a `const std::string`. Which suggests the correct solution is to make `get_freshness()` and friends to return a non-const, as well.

@jbytheway 
> that change compiles for me.

Sure, but `clang++` complains:

```
CCACHE_CPP2=1 ccache clang++  -DRELEASE -DGIT_VERSION -DLOCALIZE -ffast-math -Os  -Wall -Wextra  -g  -fsigned-char -Werror -std=c++11 -MMD -MP -m32 -I/usr/include/lua5.2 -DLUA -D_GNU_SOURCE -I/usr/include/ncursesw -c src/game_inventory.cpp -o obj/game_inventory.o
[...]
src/game_inventory.cpp:400:17: error: return type 'basic_string<[...]>' must match previous return type 'const basic_string<[...]>' when lambda expression has unspecified explicit return type
                return std::string();
                ^
1 error generated.
Makefile:787: recipe for target 'obj/game_inventory.o' failed
make: *** [obj/game_inventory.o] Error 1
```

Really, the solutions are simple: 
- have `get_freshness` return a non-const string
- de-const the `return get_freshness(...)`
- const the other `return std::string()`

I'm going to apply the first of these, please advise if you'd rather see something else.

Reply:
@BevapDin Yes, but `to_string_clipped` does not return a `const std::string`. Which suggests the correct solution is to make `get_freshness()` and friends to return a non-const, as well.

@jbytheway 
> that change compiles for me.

Sure, but `clang++` complains:

```
CCACHE_CPP2=1 ccache clang++  -DRELEASE -DGIT_VERSION -DLOCALIZE -ffast-math -Os  -Wall -Wextra  -g  -fsigned-char -Werror -std=c++11 -MMD -MP -m32 -I/usr/include/lua5.2 -DLUA -D_GNU_SOURCE -I/usr/include/ncursesw -c src/game_inventory.cpp -o obj/game_inventory.o
[...]
src/game_inventory.cpp:400:17: error: return type 'basic_string<[...]>' must match previous return type 'const basic_string<[...]>' when lambda expression has unspecified explicit return type
                return std::string();
                ^
1 error generated.
Makefile:787: recipe for target 'obj/game_inventory.o' failed
make: *** [obj/game_inventory.o] Error 1
```

Really, the solutions are simple: 
- have `get_freshness` return a non-const string
- de-const the `return get_freshness(...)`
- const the other `return std::string()`

I'm going to apply the first of these, please advise if you'd rather see something else.